### PR TITLE
docs: refresh design doc to current architecture

### PR DESCRIPTION
**Pull request by jcogilvie**
### Description of your changes

The design doc at `design/design-doc-cli-diff.md` had drifted significantly from the current implementation — most interface signatures were stale, the `comp` subcommand and its entire processor were undocumented, and the `render.Outputs` narrative referred to a struct that has since been renamed and restructured upstream. This PR brings every section, interface block, and diagram in sync with the current `cmd/diff/` tree.

It also adds a "Keeping Documentation in Sync" section and a "Git, Commits, Issues, and Pull Requests" section to `CLAUDE.md` so future code changes prompt corresponding doc updates and PRs follow project conventions (DCO sign-off, draft PRs, template compliance). The mermaid SVG regeneration recipe points at the `minlag/mermaid-cli` Docker image for portability.

What's new in the doc:

- Document the `comp` subcommand and `CompDiffProcessor` (entirely missing).
- Rewrite all interface blocks in §6 to current signatures; add `FunctionProvider`, `CompDiffRenderer`, and the structured-output types.
- Replace the obsolete `render.Outputs` / `RenderWithRequirements` narrative with the upstream `CompositionOutputs` / `RenderToStableState` reality, including `--eventual-state` mode.
- Cover nested-XR recursion, the two-phase diff calculation, claim-with-`spec.claimRef` synthesis, and `Cleanup` semantics.
- Drop §10.15 (`--namespace` flag is complete: `xr` reads namespace from YAML; `comp`'s `-n` filter has shipped).
- Add §4 test-case categories for `comp`, structured output, nested XRs, eventual state, and claim edge cases.
- Update §12 layout and §8 CLI examples for `xr` / `comp` / `--output` / etc.
- Refresh all 10 mermaid sources and regenerate SVGs.

I have:

- [x] Read and followed Crossplane's [contribution process].
- ~~Run `earthly -P +reviewable` to ensure this PR is ready for review.~~ Docs-only change; no code paths exercised.
- ~~Added or updated unit tests.~~ Docs-only change.
- ~~Added or updated e2e tests.~~ Docs-only change.
- [x] Documented this change as needed.
- ~~Followed the [API promotion workflow] if this PR introduces, removes, or promotes an API.~~ No API changes.

[contribution process]: https://github.com/crossplane/crossplane/tree/main/contributing
[cheat sheet]: https://github.com/crossplane/crossplane/tree/main/contributing#checklist-cheat-sheet
[API promotion workflow]: https://github.com/crossplane/crossplane/blob/main/contributing/guide-api-promotion.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -390,6 +390,34 @@ Two supported modes:
...
+```yaml
+jobs:
+  diff:
+    runs-on: ubuntu-latest
+    container:
+      image: my-org/crossplane-diff:latest
+    env:
+      CROSSPLANE_DIFF_DOCKER_NETWORK: ${{ job.container.network }}
+    steps:
+      - run: crossplane-diff xr xr.yaml
```
[the remark this answers is not in the document]
Comment (jcogilvie) [reply README.md:413]: this is incorrect https://docs.github.com/en/actions/reference/workflows-and-actions/contexts
